client/core/simnet_trade.go

```diff
@@ -1193,91 +1172,98 @@ func (s *simulationTest) monitorTrackedTrade(client *simulationClient, tracker *
 		client.log.Infof("updated %s balance diff with %s", a.symbol, a.valFmt(amt))
 		client.expectBalanceDiffs[a.id] += amt
 	}
+	for _, match := range tracker.matches {
+		if match.Side == order.Taker {
+			if finalStatus >= order.TakerSwapCast {
+				recordBalanceChanges(true, match.Quantity, match.Rate)
+			}
+			if finalStatus >= order.MatchComplete {
+				recordBalanceChanges(false, match.Quantity, match.Rate)
+			}
+		}
+		if match.Side == order.Maker {
+			if finalStatus >= order.MakerSwapCast {
+				recordBalanceChanges(true, match.Quantity, match.Rate)
+			}
+			if finalStatus >= order.MakerRedeemed {
+				recordBalanceChanges(false, match.Quantity, match.Rate)
+			}
+		}
+	}
+	return nil
+}
 
+func (s *simulationTest) monitorTrackedTrade(client *simulationClient, tracker *trackedTrade, finalStatus order.MatchStatus) error {
 	// run a repeated check for match status changes to mine blocks as necessary.
 	maxTradeDuration := 2 * time.Minute
 
 	tryUntil(s.ctx, maxTradeDuration, func() bool {
 		var completedTrades int
+		mineAssets := make(map[uint32]uint32)
+		// Don't spam.
+		time.Sleep(time.Second * 2 * sleepFactor)
 		tracker.mtx.Lock()
-		defer tracker.mtx.Unlock()
 		for _, match := range tracker.matches {
 			side, status := match.Side, match.Status
-			if status >= finalStatus {
-				// With async redeem request, wait for redeem sig before
-				// declaring the match complete. Sometimes it is almost
-				// instantaneos and other times the server's node takes a while.
-				// to see the txns (TODO: diagnose this).
-				// Tests that don't expect a redeem can set noRedeemWait to true.
-				if finalStatus == order.MatchComplete &&
-					len(match.MetaData.Proof.Auth.RedeemSig) == 0 &&
-					!client.noRedeemWait {
-					client.log.Infof("Completed match waiting for async redeem response...")
-					continue // check again later
-				}
-				// Prevent further action by blocking the match with a swapErr.
```

Comment:
order.MatchConfirmed now signifies the Redeem completed, so this check is not necessary.